app/services/invoices/progressive_billing_service.rb

```diff
@@ -11,15 +11,24 @@ def initialize(usage_thresholds:, lifetime_usage:, timestamp: Time.current)
     end
 
     def call
-      ActiveRecord::Base.transaction do
+      Idempotency.transaction do
         create_generating_invoice
         create_fees
         create_applied_usage_thresholds
 
+        Idempotency.unique!(invoice,
+          organization_id: lifetime_usage.organization_id,
+          external_subscription_id: subscription.external_id,
+          invoiced_usage: lifetime_usage.invoiced_usage_amount_cents,
+          threshold_amount: usage_thresholds.map(&:amount_cents).max)
```

Comment:
How does it work with recurring thresholds?

Reply:
it doesn't! I had spec failures related to it.

That's why I added:

```
credits = Credits::ProgressiveBillingService.call(invoice:).credits
        if credits.any?
          Idempotency.unique!(invoice, previous_progressive_billing_invoice_id: credits.first.progressive_billing_invoice_id)
        end
```